chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/ChaosMonkeyConfiguration.java

```diff
@@ -135,8 +140,30 @@ public MemoryAssault memoryAssault() {
 
   @Bean
   public ChaosMonkeyRequestScope chaosMonkeyRequestScope(
-      List<ChaosMonkeyRequestAssault> chaosMonkeyAssaults, List<ChaosMonkeyAssault> allAssaults) {
-    return new ChaosMonkeyRequestScope(settings(), chaosMonkeyAssaults, allAssaults, publisher());
+      List<ChaosMonkeyRequestAssault> chaosMonkeyAssaults,
+      List<ChaosMonkeyAssault> allAssaults,
+      ChaosToggles chaosToggles) {
+    return new ChaosMonkeyRequestScope(
+        settings(), chaosMonkeyAssaults, allAssaults, publisher(), chaosToggles);
+  }
+
+  @Configuration
+  @ConditionalOnClass(Unleash.class)
+  static class UnleashChaosConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnBean(Unleash.class)
+    public ChaosToggles unleashChaosToggles(
+        Unleash unleash, ChaosMonkeyProperties chaosMonkeyProperties) {
+      return new UnleashChaosToggles(chaosMonkeyProperties, unleash);
+    }
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(ChaosToggles.class)
```

Comment:
By using `ConditionalOnMissingBean` we provide a default implementation that mimics the current behavior, while allowing the user to override with others.

Reply:
Here im unsure if we need a DefaultChaosToggleNameMapper at all, as its only important if we have Unleashed enabled.

Reply:
We need a default since the calling code to check the toggle uses it: 
```java
chaosToggles.isEnabled(chaosToggleNameMapper.mapName(type, name))
```

If a user was to create their own `ChaosToggle` implementation , they would want to breakout by `ChaosTarget` that the default mapper provides.